lib/Controller/WopiController.php

```diff
@@ -671,6 +671,16 @@ public function postFile($fileId, $access_token) {
 		$wopiOverride = $this->request->getHeader('X-WOPI-Override');
 
 		if ($this->lockManager->isLockProviderAvailable()) {
+			$locks = $this->lockManager->getLocks($wopi->getFileid());
+			$existingLock = array_pop($locks);
```

Comment:
other locks would not be relevant by principle?

Reply:
There is currently just the possibility of one lock the way how files_lock app handles them. In the future multiple locks might be possible, if we support locking folders as well. I'll switch to array_shift instead as we are only interested in the closest lock to the file.

Reply:
Adjusted